bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/FloureonThermostatHandler.java

```diff
@@ -0,0 +1,260 @@
+
+    private void handleSetTimeCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof DateTimeType) {
+            ZonedDateTime zonedDateTime = ((DateTimeType) command).getZonedDateTime();
+            try {
+                new SetTimeCommand(tob(zonedDateTime.getHour()), tob(zonedDateTime.getMinute()),
+                        tob(zonedDateTime.getSecond()), tob(zonedDateTime.getDayOfWeek().getValue()))
+                                .execute(floureonDevice);
+            } catch (Exception e) {
+                logger.warn("Error while setting time of {} to {}: {}", thing.getUID(), command, e.getMessage());
+            }
+        } else {
+            logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    @Override
+    protected void refreshData() {
+        if (ThingStatus.ONLINE != thing.getStatus()) {
+            return;
+        }
+
+        FloureonDevice floureonDevice = this.floureonDevice;
+        if (floureonDevice != null) {
+            try {
+                AdvancedStatusInfo advancedStatusInfo = floureonDevice.getAdvancedStatus();
+                if (advancedStatusInfo == null) {
+                    logger.warn("Device {} did not return any data. Trying to reauthenticate...", thing.getUID());
+                    authenticate();
+                    advancedStatusInfo = floureonDevice.getAdvancedStatus();
+                }
+                if (advancedStatusInfo == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Device not responding.");
+                    return;
+                }
+                logger.trace("Retrieved data from device {}: {}", thing.getUID(), advancedStatusInfo);
+                updateState(ROOM_TEMPERATURE, new QuantityType<>(advancedStatusInfo.getRoomTemp(), SIUnits.CELSIUS));
+                updateState(ROOM_TEMPERATURE_EXTERNAL_SENSOR,
+                        new QuantityType<>(advancedStatusInfo.getExternalTemp(), SIUnits.CELSIUS));
+                updateState(SETPOINT, new QuantityType<>(advancedStatusInfo.getThermostatTemp(), SIUnits.CELSIUS));
+                updateState(POWER, OnOffType.from(advancedStatusInfo.getPower()));
+                updateState(MODE, StringType.valueOf(advancedStatusInfo.getAutoMode() ? "auto" : "manual"));
+                updateState(SENSOR, StringType.valueOf(advancedStatusInfo.getSensorControl().name()));
+                updateState(TEMPERATURE_OFFSET, new QuantityType<>(advancedStatusInfo.getDif(), SIUnits.CELSIUS));
+                updateState(ACTIVE, OnOffType.from(advancedStatusInfo.getActive()));
+                updateState(REMOTE_LOCK, OnOffType.from(advancedStatusInfo.getRemoteLock()));
+                updateState(TIME, new DateTimeType(getTimestamp(advancedStatusInfo)));
+            } catch (Exception e) {
+                logger.warn("Error while retrieving data for {}: {}", thing.getUID(), e.getMessage());
```

Comment:
If this happens you might want to set the thing to offline (and log to debug). So the user is also informed about this.

Reply:
done.